pkg/services/k8s/apiserver/admission/schema_validate.go

```diff
@@ -0,0 +1,81 @@
+package admission
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/grafana/grafana/pkg/registry/corekind"
+	"github.com/grafana/kindsys/encoding"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apiserver/pkg/admission"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+)
+
+const PluginNameSchemaValidate = "SchemaValidate"
+
+// Register registers a plugin
+func RegisterSchemaValidate(plugins *admission.Plugins, reg *corekind.Base) {
+	plugins.Register(PluginNameSchemaValidate, func(config io.Reader) (admission.Interface, error) {
+		return NewSchemaValidate(reg), nil
+	})
+}
+
+type schemaValidate struct {
+	log log.Logger
+	reg *corekind.Base
+}
+
+var _ admission.ValidationInterface = schemaValidate{}
+
+// Validate makes an admission decision based on the request attributes.  It is NOT allowed to mutate.
+func (sv schemaValidate) Validate(ctx context.Context, a admission.Attributes, o admission.ObjectInterfaces) (err error) {
+	obj := a.GetObject()
+	sv.log.Info(fmt.Sprintf("validating resource of kind %s", a.GetKind().Kind))
+	ck := sv.reg.ByName(obj.GetObjectKind().GroupVersionKind().Kind)
+	if ck == nil {
+		sv.log.Info(fmt.Sprintf("did not match %s", obj.GetObjectKind().GroupVersionKind().Kind))
```

Comment:
Shouldn't be a warn here?

Reply:
When would that happen? 
Is there any expected call to this admission with kinds not registered at `corekind.Base` (e.g. other k8s objects)?

Reply:
Good point - long term, it definitely should be, because long term...

> Is there any expected call to this admission with kinds not registered at corekind.Base (e.g. other k8s objects)?

...the admission controller will have a registry with an exhaustive list of all kinds known to the apiserver, so if it encounters a resource of a kind not in that list, it should immediately reject it.

i wrote it this way because i don't know how experimental this code is, though, so i don't know if the folks iterating on it ( @toddtreece @charandas @ryantxu ?) are ready to ratchet up the correctness checks to the point where we'd be rejecting unknown resource types.

At minimum though, i will change it to a warn

Reply:
right now this will be called for the GRDs as well. GRDs could be ignored if the group == `kinds.grafana.com`

Reply:
Doesn't that look like `kind` is empty? Is that expected? 🤔